base_layer/p2p/src/services/liveness/service.rs

```diff
@@ -327,25 +341,44 @@ where
     }
 
     async fn start_ping_round(&mut self) -> Result<(), LivenessError> {
-        let monitored_peers = { self.monitored_peers.read().await.clone() };
-        let selected_peers = self
+        let mut monitored_peers = { self.monitored_peers.read().await.clone() };
+        // Try to select connections that exclude seed nodes
+        let mut selected_peers = self
             .connectivity
             .select_connections(ConnectivitySelection::random_nodes(
                 self.config.num_peers_per_round,
-                Default::default(),
+                self.seeds.clone(),
             ))
             .await?
             .into_iter()
             .map(|c| c.peer_node_id().clone())
-            .chain(monitored_peers)
             .collect::<Vec<_>>();
 
+        // If no connections were selected, use connected seed nodes as a fallback
```

Comment:
this should not be empty. 
This should be if the `selected_peers.len()` < num_peers_per_round